config/config.go

```diff
@@ -321,30 +334,24 @@ func (cfg *config) ChangeUnlockPassword(currentUnlockPassword string, newUnlockP
 			logger.Logger.WithField("key", userConfig.Key).Info("re-encrypted key")
 		}
 
-		newSecret, err := randomHex(32)
+		// delete the JWT secret so it will be re-generated on next unlock (to log all sessions out on password change)
+		err = tx.Where(&db.UserConfig{Key: "JWTSecret"}).Delete(&db.UserConfig{}).Error
```

Comment:
we currently create the secret in the Unlock function, don't we? so it's a bit of a side effect and not explicit and easy noticeable. 

is the deletion here enough? or maybe could it be RegenerateJWTSecret() or something?

Reply:
I understand there is a side effect, but I think this is inevitable because now the JWT secret is stored decrypted in memory - and that is a side effect that we have to keep in mind. Also this code is inside a DB transaction, so the code for generating a new JWT secret had to be duplicated and slightly different than the other generation code.

I covered this quite well with tests I think. Do you think it's ok as-is?